src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SearchIndexEntryJObjectGenerator.cs

```diff
@@ -83,14 +83,31 @@ void ISearchValueVisitor.Visit(DateTimeSearchValue dateTime)
 
         void ISearchValueVisitor.Visit(NumberSearchValue number)
         {
-            AddProperty(SearchValueConstants.NumberName, number.Number);
+            if (number.Low == number.High)
```

Comment:
For maintaining backwards compatibility until we can do migration, we will want to dual write to `SearchValueConstants.NumberName` when `Low == High`.